src/Venflow/Venflow/Database.cs

```diff
@@ -63,8 +60,8 @@ protected Database(DatabaseOptionsBuilder optionsBuilder)
         {
             ConnectionString = optionsBuilder.ConnectionString;
             DefaultLoggingBehavior = optionsBuilder.DefaultLoggingBehavior;
-            _loggers = optionsBuilder.Loggers;
-
+            _loggers = optionsBuilder.Loggers.Count > 0 ? optionsBuilder.Loggers : Array.Empty<LoggerCallback>();
+            _databaseConfigurationOptionsBuilder = new DatabaseConfigurationOptionsBuilder(this.GetType());
```

Comment:
Also this should only be created once for all Database instances of that type, therefor safe its state in the `DatabaseConfiguration` class. The idea behind this is that the `DatabaseConfigurationOptionsBuilder` stores all configuration which will never change throughout the lifetime of the application, whereas the `DatabaseOptionsBuilder` is storing information which may change during runtime.